index.js

```diff
@@ -102,3 +109,43 @@ axios
     setFailed(error.message);
     endGroup();
   });
+
+let pipelineComplete = false;
+const pollInterval = 3000; // in milliseconds
+
+const keepPolling = getInput("Follow").toLowerCase();
+
+const pollWorkflow = () => {
+  axios
+    .get(workFlowUrl, {
+      headers: headers,
+    })
+    .then((response) => {
+      if (response.data.items[0].status != "running") {
+        pipelineComplete = true;
+        if (response.data.items[0].status == "success") {
+          info("CircleCI Pipeline is complete");
+        } else {
+          setFailed("CircleCI Pipeline failed");
+        }
+      }
+    })
+    .catch((error) => {
+      startGroup("Failed to Poll CircleCI Pipeline");
+      coreError(error);
+      setFailed(error.message);
+      endGroup();
+    });
+};
+
+info("Poll CircleCI Workflow is is set to: " + keepPolling);
+
+if (keepPolling == "true") {
```

Comment:
I would combine `pipelineComplete` with `keepPolling`, instead of managing two different flags.